src/deepsparse/pipeline.py

```diff
@@ -118,35 +123,39 @@ class PipelineImplementation(Pipeline):
         other runtime information that will be used across instances of the
         MultiModelEngine to provide optimal performance when running multiple
         models concurrently
-    :param threadpool: An optional ThreadPoolExecutor() object, if given the
+    :param executor: A optional ThreadPoolExecutor() object, if given the
         pipeline executes inference requests in a non-blocking manner and returns
-        a Future object, call Future.result() on returned object to get the result
+        a Future object, call Future.result() on returned object to get the result.
+        Can also accept an int number of workers, a ThreadPoolExecutor object is
+        auto-initialized with the specified integer in that case; if True a
+        ThreadPoolExecutor is initialized with min(32, (os.cpu_count() or 1) + 4)
+        max workers. False and None represent synchronous execution
     """
 
     def __init__(
         self,
         model_path: str,
         engine_type: str = DEEPSPARSE_ENGINE,
-        batch_size: int = 1,
+        batch_size: Optional[int] = 1,
```

Comment:
defaulted back to 1, to not run in dynamic batch mode, if not explicitly specified by user, to run in dynamic batch mode, user must specify `batch_size=None` when creating Pipelines